tests/meta/clean/test_clean_flow_cells_api.py

```diff
@@ -0,0 +1,246 @@
+
+    # THEN the check whether the flow cell can be deleted returns True
+    with mock.patch(
+        "cg.meta.clean.clean_flow_cells.CleanFlowCellAPI.is_directory_older_than_21_days",
+        return_value=True,
+    ):
+        assert flow_cell_clean_api_can_be_removed.can_flow_cell_directory_be_deleted()
+
+
+def test_delete_flow_cell_directory(flow_cell_clean_api_can_be_removed: CleanFlowCellAPI):
+    """Test that a flow cell directory is removed."""
+    # GIVEN a flow cell that can be removed
+
+    # GIVEN that the flow cell directory exists
+    assert flow_cell_clean_api_can_be_removed.flow_cell.path.exists()
+
+    # WHEN removing the flow cell directory
+    with mock.patch(
+        "cg.meta.clean.clean_flow_cells.CleanFlowCellAPI.is_directory_older_than_21_days",
+        return_value=True,
+    ):
+        flow_cell_clean_api_can_be_removed.delete_flow_cell_directory()
+
+    # THEN the flow cell directory is removed
+    assert not flow_cell_clean_api_can_be_removed.flow_cell.path.exists()
+
+
+def test_delete_flow_cell_directory_can_not_be_deleted(
+    flow_cell_clean_api_can_not_be_removed: CleanFlowCellAPI,
+):
+    """Test delete a flow cell that does not pass all checks."""
+    # GIVEN a flow cell that should not be removed.
+
+    # GIVEN that the flow cell directory exists
+    assert flow_cell_clean_api_can_not_be_removed.flow_cell.path.exists()
+
+    # WHEN trying to remove the flow cell
+    with pytest.raises(CleanFlowCellFailedError):
+        flow_cell_clean_api_can_not_be_removed.delete_flow_cell_directory()
+
+    # THEN the flow cell directory still exists
+    flow_cell_clean_api_can_not_be_removed.flow_cell.path.exists()
+
+
+def test_get_flow_cell_from_statusdb_does_not_exist(
+    flow_cell_clean_api_can_not_be_removed: CleanFlowCellAPI,
+):
+    """Test retrieving a flow cell from statusDB that does not exist."""
+    # GIVEN a CleanFlowCellAPI with a flow cell that is not in statusDB
+    flow_cell_clean_api_can_not_be_removed.flow_cell.id = "flow_cell_does_not_exist"
+
+    assert not flow_cell_clean_api_can_not_be_removed.status_db.get_flow_cell_by_name(
+        flow_cell_clean_api_can_not_be_removed.flow_cell.id
+    )
+
+    # WHEN retrieving the flow cell from statusDB
+
+    # THEN a ValueError is raised
+    with pytest.raises(ValueError):
+        flow_cell_clean_api_can_not_be_removed.get_flow_cell_from_status_db()
+
+
+def test_flow_cell_has_not_fastq_files_in_housekeeper(
+    flow_cell_clean_api_can_not_be_removed: CleanFlowCellAPI, store_with_flow_cell_to_clean: Store
+):
+    """Test to check wether a flow cell has files in Housekeeper."""
+    # GIVEN a flow cell that has entries in StatusDB but does not have any files in Housekeeper for its samples
+    flow_cell_clean_api_can_not_be_removed.status_db = store_with_flow_cell_to_clean
+
+    # WHEN checking whether a sample on a flow cell has fastq files
+
+    # THEN a HousekeeperBundleVersionMissingError is raised
```

Comment:
Adding a caplog assertion would give more value to this test, specially because this error can be caused by many different issues